blackbox_solver/src/lib.rs

```diff
@@ -0,0 +1,281 @@
+#![warn(unused_crate_dependencies)]
```

Comment:
Can you give this a crate level doc comment, explaining its usage with //!

Reply:
sure!